src/Symfony/Component/Validator/Constraints/MacAddressValidator.php

```diff
@@ -43,11 +70,49 @@ public function validate(mixed $value, Constraint $constraint): void
             $value = ($constraint->normalizer)($value);
         }
 
-        if (!filter_var($value, \FILTER_VALIDATE_MAC)) {
+        if (!self::checkMac($value, $constraint->type)) {
             $this->context->buildViolation($constraint->message)
                 ->setParameter('{{ value }}', $this->formatValue($value))
                 ->setCode(MacAddress::INVALID_MAC_ERROR)
                 ->addViolation();
         }
     }
+
+    /**
+     * Checks whether an MAC address is unicast or multicast.
```

Comment:
```suggestion
     * Checks whether a MAC address is unicast or multicast.
```

Can you make the same change everywhere?